src/modules/paymentProcessor/bridging/PP_Connext_Crosschain_v1.sol

```diff
@@ -0,0 +1,58 @@
+pragma solidity ^0.8.20;
+
+import {CrosschainBase_v1} from
+    "src/modules/paymentProcessor/bridging/abstracts/CrosschainBase_v1.sol";
+import {ICrossChainBase_v1} from
+    "src/modules/paymentProcessor/bridging/abstracts/ICrosschainBase_v1.sol";
+import {ConnextBridgeLogic} from "./ConnextBridgeLogic.sol";
+import {IERC20PaymentClientBase_v1} from
+    "@lm/interfaces/IERC20PaymentClientBase_v1.sol";
+
+contract PP_Connext_Crosschain_v1 is CrosschainBase_v1 {
+    ConnextBridgeLogic public connextBridgeLogic;
+
+    constructor(uint chainId_, address connextBridgeLogic_)
```

Comment:
```suggestion
 /// @inheritdoc Module_v1
    function init(
        IOrchestrator_v1 orchestrator_,
        Metadata memory metadata,
        bytes memory configData
    ) external override(Module_v1) initializer {
        __Module_init(orchestrator_, metadata);

        (uint chainId_, address connextBridgeLogic_) = abi.decode(configData, (uint, address));
        
        // set chainId
        
          connextBridgeLogic = ConnextBridgeLogic(connextBridgeLogic_);
    }

```

Reply:
The modules are upgradeable so they should work through initialization. Also each module gets its parameters through the `bytes configdata` in the init. So in this case the `Module_v1` init is overridden as it is inherited in the `CrosschainBase_v1`

Reply:
Ahhh okay this is nice catch as well, didn't realize this but yes we should be using init and not the constructor. @zzzuhaibmohd please see this as you are making changes in the other PR and this is relevant.

Reply:
Resolved this in the new changes.

Reply:
@FHieser I need some help on understanding this `PP_Connext_Crosschain_v1.sol` inherits 'PP_Crosschain' which then inherits 'CrosschainBase_v1'. How do the initializers work here especially when dealing with this nested inheritance pattern?

Reply:
Yes that is a really good question. Took me a bit to solve it too to solve it in this case.
First what you have to do to solve it:
CrosschainBase -> Remove init
PP_CrossChain -> Remove init
PP_Connext_Crosschain -> Use `override(Module_v1) ` (You'll have to import it)

Reply:
Now to the explanation, starting with the bad part:
The initialization concept we currently use straight up wasnt properly designed for multi contract inheritance 😅
The initial thought process that one might follow is, that we have to pass along the configdata through the inheritance tree until it lands at the correct place. But as we dont have a standard defined here, of how this would actually looks like, this will get confusing very fast. We initially solved this by creating a different __Module_init, that can be called separately, but this solution is probably not ideal too. So essentially, you stumbled upon something that we didnt define properly, which we will do.
Now to the good part:
We are lucky in this case, that only the Connext_Crosschain actually needs to set data during initialization, which the solution that I posted above basically uses. We just say ignore all the abstract contracts, as they dont need any initialization data in the first place. 
Nice to get lucky I guess, but this isnt feasable in the long run. I will get together with my colleagues, to get a proper concept together, hopefully this week.

Reply:
Awesome thanks! Well glad that us running into these things is helping you guys on the other end. I'm curious if the other team will also run into a similar issue. Will start a discussion in the thread around initialization and we can discuss it there as well. Thanks for the response here. Also Zuhaib fixed this so Im resolving here.